src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -126,7 +127,7 @@ CHIP_ERROR IcdManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoi
 
                 Structs::MonitoringRegistrationStruct::Type s{ .checkInNodeID    = e.checkInNodeID,
                                                                .monitoredSubject = e.monitoredSubject,
-                                                               .key              = e.key,
+                                                               .key              = chip::ByteSpan(), // Do not expose the key
```

Comment:
Instead of returning an empty byte span each time, we should remove this entirely from the read.

Reply:
Will do in a follow up once the specs PR is merged. issue to track is #29288